### PR TITLE
IPv6: Allow AAAA record use

### PR DESCRIPTION
**Pull request by pmichali**
By default, DNS is set up to ignore (IPv6) AAAA records. As a result,
whenever a remote site is accessed, only the (IPv4) A record is used,
and a synthesized IPv6 address formed. This means that NAT64 will be
used for all external accesses, even when a site supports IPv6.

This is fine for environments where the cluster only has IPv4 access,
like GCE currently. However, if the cluster will have IPv6 access too,
the configuration can be modified to use AAAA records in DNS lookups,
and then remote sites with IPv6 addresses can be directly accessed
(without using NAT64).

This change provides a knob to allow enabling use of AAAA records. In
addition to modifying the DNS configuration, ip6tables rules are
created to perform SNAT for packets coming from the container nodes
and to forward external packets to the kubeadm-dind-net to get to the
pods.

The boolean environment variable, ALLOW_AAAA_USE, is defined for this
capability. The default is NOT to allow use of AAAA records, for
backwards compatibility.

Fixes issue #147

### CODE REVIEW

File: dind-cluster.sh [hunk 1]
```diff
@@ -76,6 +76,7 @@ if [[ ${IP_MODE} = "ipv6" ]]; then
     dns_server="${dind_ip_base}100"
     DEFAULT_POD_NETWORK_CIDR="fd00:10:20::/72"
     USE_HAIRPIN="${USE_HAIRPIN:-true}"  # defaults on
+    ALLOW_AAAA_USE="${ALLOW_AAAA_USE:-false}"  # Default is to use DNS64 always
```
Comment (rpothier) [inline dind-cluster.sh]: Would it make sense going forward that we prefix new env variables with DIND_?
I know only some have the prefix at this point, (DIND_SUBNET, DIND_IMAGE ...) but it makes it easier to review the env variables.

Comment (pmichali) [reply dind-cluster.sh]: Sure, will update.

**Pull request merged**